acceptance/analyzer_test.go

```diff
@@ -145,34 +127,57 @@ func setupAnalyzeFixtures(t *testing.T) analyzeFixtures {
 
 	// Registry
 
+	// setup read and write images
 	someAppName := "some-app-image-" + h.RandString(10)
 	fixtures.authRegAppImage, fixtures.authRegAuthConfig = buildAuthRegistryImage(
 		t,
 		someAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
-	fixtures.readOnlyRegAppImage = readOnlyRegistry.RepoName(someAppName)
+	testRegistry.Add(someAppName, ih.NewImagePrivileges(readWriteImage))
 
 	someOtherAppName := "some-other-app-image-" + h.RandString(10)
 	fixtures.authRegOtherAppImage, _ = buildAuthRegistryImage(
 		t,
 		someOtherAppName,
 		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
 		"--build-arg", "fromImage="+containerBaseImage,
 		"--build-arg", "metadata="+appMeta,
 	)
+	testRegistry.Add(someOtherAppName, ih.NewImagePrivileges(readWriteImage))
 
-	someCacheName := "some-cache-image-" + h.RandString(10)
+	// setup read only images
+	someReadOnlyAppName := "some-readonly-app-image-" + h.RandString(10)
+	fixtures.readOnlyRegAppImage, _ = buildAuthRegistryImage(
+		t,
+		someReadOnlyAppName,
+		filepath.Join("testdata", "analyzer", "app-image"),
+		testRegistry,
+		"--build-arg", "fromImage="+containerBaseImage,
+		"--build-arg", "metadata="+appMeta,
+	)
+	testRegistry.Add(someReadOnlyAppName, ih.NewImagePrivileges(onlyReadImage))
+
+	someCacheName := "some-cache-image"
```

Comment:
Can we keep the random string?

Reply:
Yeap

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/685/commits/27c678fc98ac3d25a2f3c0727ebff90aa0640ba1)